.github/workflows/main.yml

```diff
@@ -231,64 +245,103 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      # Ref: https://github.com/actions/checkout/issues/1590#issuecomment-2567109195
+      - name: Prepare container (Linux, containerized)
+        if: runner.os == 'Linux' && matrix.container != ''
+        run: |
+          docker pull ${{ matrix.container }}
+          docker run --name build-container -d -v ${{ github.workspace }}:/workspace ${{ matrix.container }} tail -f /dev/null
+
+          cat <<'EOF' > container-env.sh
+          export CTEST_OUTPUT_ON_FAILURE="${{ env.CTEST_OUTPUT_ON_FAILURE }}"
+          export CTEST_PARALLEL_LEVEL="${{ env.CTEST_PARALLEL_LEVEL }}"
+          export CMAKE_GENERATOR="${{ env.CMAKE_GENERATOR }}"
+          export ASAN_OPTIONS="${{ env.ASAN_OPTIONS }}"
+          export TSAN_OPTIONS="${{ env.TSAN_OPTIONS }}"
+          EOF
+
       - name: Install (Linux)
-        if: runner.os == 'Linux'
+        if: runner.os == 'Linux' && matrix.container == ''
```

Comment:
Switching all Linux build configurations over to the containerized approach would be trivial and reduce the complexity of the CI configuration if this is preferred by the maintainers.